specs/agents/tracing-spans-service-target.md

```diff
@@ -104,29 +104,32 @@ This specification assumes that values for `span.type` and `span.subtype` fit th
 - `span.context.service.target.type` should have the same value as `span.subtype` and fallback to `span.type`.
 - `span.context.service.target.name` depends on the span context attributes
 
-On agents, the following algorithm should be used to infer the values for `span.context.service.target.*` fields. 
+On agents, the following algorithm should be used to infer the values for `span.context.service.target.*` fields.
+
 ```javascript
 // span created on agent
 span = {};
 
 if (span.isExit) {
   service_target = span.context.service.target;
-  
-  if (!service_target.type) { // infer type from span type & subtype
-      
-    // use sub-type if provided, fallback on type othewise
+
+  if ('type' in service_target) { // If not manually specified, infer type from span type & subtype.
```

Comment:
Q1: @SylvainJuge If the user manually calls either of the following two equivalents:

```js
span.setDestinationService(resource='my-backend-service-name')
span.setServiceTarget(type='', name='my-backend-service-name')
```

Is the intent to have a span like this?

```json
{
    "id": "...",
    "name": "...",
    "type": "...",
    "subtype": "...",
    // ...
    "context": {
        "service": {
            "target": {
                "type": "",
                "name": "my-backend-service-name"
            }
        },
        "destination": {
            "service": {
                "name": "",
                "type": ""
                "resource": "my-backend-service-name",
            }
        }
    }
}
```

If so, then `if (!service_target.type)` is misleading. At least in JavaScript if `service.target.type` is the empty string then that if-block will be executed.  My diff here changes to use an existence check, if that is the intention.

(Same for the `if ('name' in service_target) {` change below.

Reply:
The meaning I tried to convey here was "service_target.type is not set", I wasn't aware of this subtle JS behavior.

I would however add that in this case the `span.setServiceTarget(...)` method is not meant to accept an empty `type`, only the legacy `setDestinationService` should allow that. The underlying idea is that we should _at least_ have the `service.target.type` and maybe an optional `service.target.name` to further refine that.

Reply:
After re-reading this I forgot that we added the meaning of "empty type" to make the apm-server use the `resource` value as-is, so this might actually be a valid use-case even if providing a `type` is definitely recommended (in that case the apm-server can't infer any value).

Reply:
> The meaning I tried to convey here was "service_target.type is not set"

Okay, I take this to mean that changing the spec's pseudo-code to `if ('type' in service_target) {` is reasonable to clarify.